ansible/files/bootstrap/consul/post/test/vault-integration.yml

```
@@ -0,0 +1,78 @@
+# test admin.team.policy
+#TODO: Use consul client for this??
+- name: "{{ service }} - post/test/vault-integration - admin token - write key-value to root '/' is forbidden"
+  uri:
+    # curl -v -X PUT --header "X-Consul-Token: <token>" -d 'my-admin-value' http://127.0.0.1:8500/v1/kv/my-admin-key
+    url:  http://127.0.0.1:8500/v1/kv/my-admin-key
+    method: PUT
+    body: "my-admin-value"
+    status_code: 403
+    return_content: yes
+    headers:
+      X-Consul-Token: "{{ consul.acl.admin_team.token }}"
+
+- name: "{{ service }} - post/test/vault-integration - admin token - write key-value to /team/admin is allowed"
+  uri:
+    # curl -v -X PUT --header "X-Consul-Token: <token>" -d 'my-admin-value' http://127.0.0.1:8500/v1/kv/team/admin/my-admin-key
+    url:  http://127.0.0.1:8500/v1/kv/team/admin/my-admin-key
+    method: PUT
+    body: "my-admin-value"
+    status_code: 200
+    return_content: yes
+    headers:
+      X-Consul-Token: "{{ admin_team_token.stdout }}"
+
+- name: "{{ service }} - post/test/vault-integration - admin token - list all the consul ACL policies is allowed"
+  uri:
+    # curl --header "X-Consul-Token: <token>" http://127.0.0.1:8500/v1/acl/policies
+    url: http://127.0.0.1:8500/v1/acl/policies
+    method: GET
+    headers:
+      X-Consul-Token: "{{ admin_team_token.stdout }}"
+    status_code: 200
+    return_content: yes
+    body_format: json
+  register: result_policies
+
+# Test dev.team.policy
+- name: "{{ service }} - post/test/vault-integration - dev token - write key-value to root '/' is forbidden"
+  uri:
+    # curl -v -X PUT --header "X-Consul-Token: <token>" -d 'my-dev-value' http://127.0.0.1:8500/v1/kv/my-dev-key
+    url:  http://127.0.0.1:8500/v1/kv/my-dev-key
+    method: PUT
+    body: "my-dev-value"
+    status_code: 403
+    return_content: yes
+    headers:
+      X-Consul-Token: "{{ dev_team_token.stdout }}"
```

```suggestion
      X-Consul-Token: "{{ consul.acl.dev_team.token }}"
```

This change needs to be made in the rest of the ansible tasks that make use of **dev_team_token**.

---

I am not sure I understand..

---

I do not see that this variable **dev_team_token** gets registered anywhere in this file: [/post/test/vault-integration.yml](https://github.com/fredrikhgrelland/vagrant-hashistack/blob/29c5fb2cd00ae89159d547504c3cc313b727b7c7/ansible/files/bootstrap/consul/post/test/vault-integration.yml) so not sure how this ansible task runs without failing!
I took a closer look at the [test run logs](https://github.com/fredrikhgrelland/vagrant-hashistack/runs/1248862811?check_suite_focus=true#step:9:596) from github checks and i do not see tasks from this file being run, i guess this file needs to be included in [810-tests.yml](https://github.com/fredrikhgrelland/vagrant-hashistack/blob/29c5fb2cd00ae89159d547504c3cc313b727b7c7/ansible/files/bootstrap/consul/post/810-tests.yml) 

So, basically I see 3 issues:

1. Tasks from this file [/post/test/vault-integration.yml](https://github.com/fredrikhgrelland/vagrant-hashistack/blob/29c5fb2cd00ae89159d547504c3cc313b727b7c7/ansible/files/bootstrap/consul/post/test/vault-integration.yml) are not run.

2. **dev_team_token** used in the ansible tasks is not registered so the ansible tasks will fail

3. Tests should be run at the end, after setting the facts so that the tests could make use of the facts that are set and so you could skip registering extra variables. Basically the sequence of ansible files in the **post** step needs to be such that it runs test at last.

Let me know if something is unclear...it might could also be that I have misunderstood the implementation so kindly ignore the above comments :)

---

WOW. Not sure how I missed this. You were right on two out of three.
1. The tests were not run, as we had not included them.
2. The variable did not exist.
3. The variables **are** set in (https://github.com/fredrikhgrelland/vagrant-hashistack/pull/386/files#diff-e0f9fbc0063fa5643b1e542f2db5b39ea2cfdddc2b8971c7eb3bc1d5a311a062R75-R79) 👍 

Changes in: 10ad0c5345b476431c59a1f68ea7f032ef7fa9e1
Thank you so much!